extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/JpaJandexScavenger.java

```diff
@@ -121,70 +122,86 @@ private static void enlistExplicitClasses(IndexView index, JpaEntitiesBuildItem
                 domainObjectCollector.addEntity(className);
             }
         }
+
+        throwIfUnIndexedClassesNotEmpty(unIndexedClasses);
     }
 
     private static void enlistReturnType(IndexView index, JpaEntitiesBuildItem domainObjectCollector,
             Set<String> enumTypeCollector) {
         Collection<AnnotationInstance> annotations = index.getAnnotations(EMBEDDED);
-        if (annotations != null && annotations.size() > 0) {
-            for (AnnotationInstance annotation : annotations) {
-                AnnotationTarget target = annotation.target();
-                DotName jpaClassName = null;
-                switch (target.kind()) {
-                    case FIELD:
-                        // TODO could fail if that's an array or a generic type
-                        jpaClassName = target.asField().type().name();
-                        break;
-                    case METHOD:
-                        // TODO could fail if that's an array or a generic type
-                        jpaClassName = target.asMethod().returnType().name();
-                        break;
-                    default:
-                        throw new IllegalStateException("[internal error] @Embedded placed on a unknown element: " + target);
-                }
-                addClassHierarchyToReflectiveList(index, domainObjectCollector, enumTypeCollector, jpaClassName);
+        if (annotations == null) {
+            return;
+        }
+
+        Set<DotName> unIndexedClasses = new HashSet<>();
```

Comment:
I would make it `unindexedClasses` as in `ReflectiveHierarchyStep`. Let's be consistent considering we follow the same pattern.

Reply:
Sold. :-)